python/nano/src/bigdl/nano/tf/keras/Model.py

```diff
@@ -13,11 +13,92 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-
-
+from time import time
 import tensorflow as tf
+import numpy as np
 
 
 class Model(tf.keras.Model):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
+
+    def fit(self,
+            x=None,
+            y=None,
+            batch_size=None,
+            epochs=1,
+            verbose='auto',
+            callbacks=None,
+            validation_split=0.,
+            validation_data=None,
+            shuffle=True,
+            class_weight=None,
+            sample_weight=None,
+            initial_epoch=0,
+            steps_per_epoch=None,
+            validation_steps=None,
+            validation_batch_size=None,
+            validation_freq=1,
+            max_queue_size=10,
+            workers=1,
+            use_multiprocessing=False,
+            perf_tune=False):
+        if perf_tune:
+            if batch_size is None:
+                batch_size = 32
+            # Data pipeline
+            print("Prefetch Datapipeline")
+            assert issubclass(
+                type(x), tf.data.Dataset), "Data object currently only support tf.data.dataset"
+
+            batched_data=[]
+            batched_target = []
+            for i in range(batch_size):
+                batched_data_item, batched_target_item = next(iter(x))
+                # batched_data.append((batched_data_item, batched_target_item))
+                batched_data.append(batched_data_item)
+                batched_target.append(batched_target_item)
+            start = time()
+            super(Model, self).fit(
```

Comment:
Could you restructure the code to make it easy to understand?
Such as writing the logic in a loop. Inside the loop, you first generate the modified arguments and then call `.fit` using the generated arguments. So that we only see this `super(Model, self).fit` once in the source code. (The DRY principle)